clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -1095,6 +1095,8 @@ class StructAccessBase
     return Visit(E->getBase());
   }
   const Expr *VisitCastExpr(const CastExpr *E) {
+    if (E->getCastKind() == CK_LValueToRValue)
+      return E;
```

Comment:
Should this be `return IsExpectedRecordDecl(E) ? E : nullptr;`?

Reply:
It might be a good sanity check, though I think I've only seen an `LValueToRValue` cast before the node we wanted.

Reply:
See https://github.com/llvm/llvm-project/pull/111448